types/metadataV13.go

```diff
@@ -0,0 +1,425 @@
+package types
+
+import (
+	"fmt"
+	"hash"
+	"strings"
+
+	"github.com/centrifuge/go-substrate-rpc-client/v3/scale"
+	"github.com/centrifuge/go-substrate-rpc-client/v3/xxhash"
+)
+
+type MetadataV13 struct {
+	Modules   []ModuleMetadataV13
+	Extrinsic ExtrinsicV11
+}
+
+func (m *MetadataV13) Decode(decoder scale.Decoder) error {
+	err := decoder.Decode(&m.Modules)
+	if err != nil {
+		return err
+	}
+	return decoder.Decode(&m.Extrinsic)
+}
+
+func (m MetadataV13) Encode(encoder scale.Encoder) error {
+	err := encoder.Encode(m.Modules)
+	if err != nil {
+		return err
+	}
+	return encoder.Encode(m.Extrinsic)
+}
+
+func (m *MetadataV13) FindCallIndex(call string) (CallIndex, error) {
+	s := strings.Split(call, ".")
+	for _, mod := range m.Modules {
+		if !mod.HasCalls {
+			continue
+		}
+		if string(mod.Name) != s[0] {
+			continue
+		}
+		for ci, f := range mod.Calls {
+			if string(f.Name) == s[1] {
+				return CallIndex{mod.Index, uint8(ci)}, nil
+			}
+		}
+		return CallIndex{}, fmt.Errorf("method %v not found within module %v for call %v", s[1], mod.Name, call)
+	}
+	return CallIndex{}, fmt.Errorf("module %v not found in metadata for call %v", s[0], call)
+}
+
+func (m *MetadataV13) FindEventNamesForEventID(eventID EventID) (Text, Text, error) {
+	for _, mod := range m.Modules {
+		if !mod.HasEvents {
+			continue
+		}
+		if mod.Index != eventID[0] {
+			continue
+		}
+		if int(eventID[1]) >= len(mod.Events) {
+			return "", "", fmt.Errorf("event index %v for module %v out of range", eventID[1], mod.Name)
+		}
+		return mod.Name, mod.Events[eventID[1]].Name, nil
+	}
+	return "", "", fmt.Errorf("module index %v out of range", eventID[0])
+}
+
+func (m *MetadataV13) FindStorageEntryMetadata(module string, fn string) (StorageEntryMetadata, error) {
```

Comment:
same as above. Why not extend 13 from 12 and reuse those methods ?

Reply:
@vedhavyas V13 is not just addition to V12, the storagemetadata enum changes.

Reply:
cool.